airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -285,13 +288,13 @@ def _read_full_refresh(
                 if self._limit_reached(internal_config, total_records_counter):
                     return
 
-    def _checkpoint_state(self, stream, stream_state, connector_state):
+    @staticmethod
+    def _checkpoint_state(stream: Stream, stream_state, state_manager):
         try:
-            connector_state[stream.name] = stream.state
+            state_manager.update_state_for_stream(stream.name, stream.namespace, stream.state)
         except AttributeError:
-            connector_state[stream.name] = stream_state
-
-        return AirbyteMessage(type=MessageType.STATE, state=AirbyteStateMessage(data=connector_state))
+            state_manager.update_state_for_stream(stream.name, stream.namespace, stream_state)
```

Comment:
It was hard to catch the difference between `stream.state` and `stream.state` :) could you please add a comment to explain in which situation the `AttributeError` is raised?